Hash Table/Partition_string.py

```diff
@@ -0,0 +1,39 @@
+def is_palindrome(s):
+    return s == s[::-1]
+
+def min_partition(s):
+    n = len(s)
+    
+    # Create a hash table to store the results of subproblems
+    # dp[i] represents the minimum number of partitions in s[0:i]
+    dp = [float('inf')] * (n + 1)
+    dp[0] = 0
+    
+    # Iterate through the string
+    for i in range(1, n + 1):
+        # Check all possible substrings s[j:i]
+        for j in range(i):
+            # If s[j:i] is a palindrome, update the minimum number of partitions
+            if is_palindrome(s[j:i]):
+                dp[i] = min(dp[i], dp[j] + 1)
+    
+    # Return the minimum number of partitions for the entire string
+    return dp[n] - 1
+
+# Example usage
+string = "aabba"
+optimal_partitions = min_partition(string)
+print("Optimal number of partitions:", optimal_partitions)
+
+
+'''
```

Comment:
kindly match the [structure ](https://github.com/akgmage/data-structures-and-algorithms/blob/main/CONTRIBUTING.md#example-program)

- move this to top
- add question
- sample io
- explanation
- time space